Source/Api/Controllers/AuthController.cs

```diff
@@ -183,20 +196,27 @@ public class AuthController : ExceptionlessApiController {
                 }
             }
 
+	        if (Settings.Current.EnableActiveDirectoryAuth && !IsAdLoginValid(model.Email, model.Password)) {
+				_logger.Error().Message("Signup failed (bad AD login) for \"{0}\"", model.Email).Tag("Signup").Identity(model.Email).SetActionContext(ActionContext).Write();
+		        return BadRequest("Username and password combination failed to authenticate with Active Directory.");
```

Comment:
I'd just return bad request with no message, otherwise you are leaking more info to a potential attacker